detox/runners/jest/DetoxAdapterImpl.js

```diff
@@ -38,12 +38,16 @@ class DetoxAdapterImpl {
   }
 
   testComplete({status, timedOut}) {
-    const _test = {
-      ...this._currentTest,
-      status,
-      timedOut,
-    };
-    this._enqueue(() => this._afterEach(_test));
+    if (this._currentTest) {
```

Comment:
If outer hooks (e.g.: `beforeAll`) fail with a timeout, we still get `afterEach` called in Detox. :scream: 
Now, I am making sure we call `detox.afterEach()` only for those tests that actually ran.